src/averageAges.js

```diff
@@ -55,7 +62,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child =>
+    onlyWithSon
+      ? child.sex === 'm' && people.some(woman => woman.name === child.mother)
+      : people.some(woman => woman.name === child.mother)
+  );
+
+  return children.reduce((total, child) => {
+    const yearOfMothersBirth = people.find(person =>
+      person.name === child.mother).born;
+
+    return (total + (child.born - yearOfMothersBirth) / children.length);
+  }, 0);
+}
+
+function getAverageAge(people) {
+  const ages = people.map(person => person.died - person.born);
+  const averageAges = ages.reduce((total, age) => total + age);
```

Comment:
The second argument for the reduce function is missing, which should be the initial value of the accumulator.

Reply:
Thank you for your response.

On [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce#:~:text=was%20called%20upon.-,initialValue,-Optional) I found, that:

> If initialValue is not specified, accumulator is initialized to the first value in the array, and callbackFn starts executing with the second value in the array as currentValue.

But, probably, in the case of readability and consistency (in another reduces I am using initial values), I should use “, 0”.